test.py

```diff
@@ -0,0 +1,240 @@
+import pytest
+from unittest.mock import MagicMock, patch
+from pathlib import Path
+
+from summarization_script import (
+    chunk_text,
+    clean_text,
+    extract_sections_from_txt,
+    summarize_with_transformer,
+    summarize_sections_single_paragraph,
+    process_folder,
+)
+
+# --- Fixtures ---
+
+@pytest.fixture
+def mock_summarizer():
+    """
+    Provides a mock summarization pipeline that always returns a fixed summary.
+    Used to isolate tests from the actual transformer model for speed and consistency.
+    """
+    mock = MagicMock()
+    mock.return_value = [{"summary_text": "This is a summary."}]
+    return mock
+
+
+@pytest.fixture
+def sample_text():
+    """
+    Sample multi-section text used to test section extraction and summarization.
+    Contains typical document sections like Introduction, Methods, Results, etc.
+    """
+    return """
+    Sections:
+    - Introduction: This is the introduction section. It provides an overview of the document.
+    - Methods: The methods section explains the methodology used in the study.
+    - Results: The results section presents the findings of the study.
+    - References: This section lists references.
+    - Acknowledgements: This section thanks contributors.
+    - Conclusion: This is the conclusion section summarizing the key points.
+    """
+
+
+# --- chunk_text tests ---
+
+def test_chunk_text():
+    """
+    Tests that chunk_text splits input text into chunks correctly based on max tokens.
+    Verifies that all sentences are present in the combined chunks.
+    """
+    text = "This is sentence one. This is sentence two. This is sentence three."
+    chunks = chunk_text(text, max_tokens=5)
+    combined = " ".join(chunks)
+    assert "sentence one" in combined
+    assert "sentence two" in combined
+    assert "sentence three" in combined
+    assert len(chunks) >= 1
+
+
+def test_chunk_text_empty():
+    """
+    Tests chunk_text behavior on empty input.
+    Expected to return a list with an empty string.
+    """
+    chunks = chunk_text("", max_tokens=5)
+    assert chunks == [""]
```

Comment:
```suggestion
    assert chunks == []
```
This is failing because it expects the function to return [""] but the function actually returns a blank list [ ] which is actually what we want so it should be `assert chunks == [ ]`